Dockerfile

```diff
@@ -1,32 +1,41 @@
-FROM ocaml/opam:alpine-3.20-ocaml-5.2 as build
+FROM ocaml/opam:alpine-3.20-ocaml-5.2 AS build
 
 # Install system dependencies
-RUN sudo apk update && sudo apk add --update libev-dev openssl-dev gmp-dev oniguruma-dev inotify-tools curl-dev autoconf
+RUN apk -U upgrade --no-cache && apk add --no-cache \
+    autoconf \
+    curl-dev \
+    gmp-dev \
+    inotify-tools \
+    libev-dev \
+    oniguruma-dev \
+    openssl-dev
 
 # Branch freeze was opam-repo HEAD at the time of commit
-RUN cd opam-repository && git pull origin c45f5bab71d3589f41f9603daca5acad14df0ab0 && opam update
+RUN cd ~/opam-repository && git fetch -q origin master && git reset --hard c45f5bab71d3589f41f9603daca5acad14df0ab0 && opam update
```

Comment:
This has always confused me. I think we've changed this line a couple of times. Back and forth between what we have now and what you're suggesting. Your line downloads some more doesn't it? Is there another difference?

Reply:
In my view, here, to avoid docker cache miss, we should just do `git checkout` to a hash that is known to be already in opam-repo history.

Reply:
I don't think it downloads more, as per the docs:

> More precisely, `git pull` runs `git fetch` with the given parameters and then depending on configuration options or command line flags, will call either `git rebase` or `git merge` to reconcile diverging branches.

So here by splitting `git pull` into `git fetch` and `git reset`, we're avoiding conflict resolution, though it's not clear to me why there would be any conflicts in the first place. Corruption could happen.

The base image already ships the opam repository, I tend to set this hash to the one already present in the image, so that the `git reset` operation is mostly free. It can be needed from time to time to update to a newer opam-repository, or just set it explicitly for reproducibility/debugging purposes. The base images are updated weekly, I don't know how often the deployed image is.

Reply:
The deployed image is updated fairly frequently. Several times a week, often more. If we can take advantage of this, it would be nice. That's why we are considering using a docker image pin instead of an opam repo pin

Reply:
Both may be needed, as the docker image is updated weekly, you might want to rely on a dependency that's been released during the week but not yet available in the base image.

Reply:
No we don't want to do that :-) We're laggards. We want to use old stuff that has been around for some time, preferably a long time. We run away from anything new, fresh, fancy or cool :-)

Reply:
There might be new, fresh, fancy, or cool bug fixes!